app/models/ingredient_name.rb

```diff
@@ -0,0 +1,5 @@
+class IngredientName < ActiveRecord::Base
+  belongs_to :ingredients
+
+  validates_uniqueness_of :recipe_ingredient_sub_name, scope: :ingredient_id
```

Comment:
* Why not validates : recipe_ingredient_sub_name, uniqueness: true ?

* if IngredientName belongs to Ingredient, why do you need to validate uniqueness of ingredient_id?